packages/react-dev-utils/WebpackDevServerUtils.js

```diff
@@ -152,9 +159,32 @@ function createCompiler(webpack, config, appName, urls, useYarn) {
     // them in a readable focused way.
     // We only construct the warnings and errors for speed:
     // https://github.com/facebook/create-react-app/issues/4492#issuecomment-421959548
-    const messages = formatWebpackMessages(
-      stats.toJson({ all: false, warnings: true, errors: true })
-    );
+    const statsData = stats.toJson({ all: false, warnings: true, errors: true });
+
+    if (useTypeScript && statsData.errors.length === 0) {
+      process.stdout.write(
```

Comment:
If we keep this can we change to use `console.log` to be consistent with other logging in this file?

Reply:
Well, the idea is to remove this line when TS finish the type-checking. I'm not sure how to do it with `console.log`. Any idea?

Reply:
We can call `clearConsole` once again but with this approach, it is much easier to modify the code later as this `if` can basically clear after itself without loosing any other important text in terminal.